native/src/main/scala/sqs4s/auth/SystemF.scala

```diff
@@ -0,0 +1,24 @@
+package sqs4s.auth
+
+import cats.effect.Sync
+import sqs4s.auth.errors.{NoEnvironmentVariablesFound, NoSystemPropertiesFound}
+
+object SystemF {
+  def env[F[_]: Sync](name: String): F[String] =
+    Sync[F].fromOption(
+      Option(System.getenv(name)),
+      NoEnvironmentVariablesFound(name)
+    )
```

Comment:
This isn't suspending the side effect of reading from the environment.
It should be:

```scala
envOpt[F](name).flatMap { opt =>
  Sync[F].fromOption(opt, NoEnvironmentVariablesFound(name))
}
```